frontend/components/events/create/forms/BasicForm.tsx

```diff
@@ -61,41 +70,59 @@ export function BasicInformation({
       updateField({ isPrivate: true });
     }
   };
-  const handleDateChange = (selectedDate: string) => {
-    // Validate if the selected date is in the past
-    const currentDate = new Date();
-    const selectedDateObj = new Date(selectedDate);
-
-    if (selectedDateObj < currentDate) {
-      setDateWarning("Selected date is in the past.");
-    } else {
-      setDateWarning(null);
-    }
+  const handleStartDateChange = (selectedDate: string) => {
+    updateField({ startDate: selectedDate });
+  };
 
-    updateField({ date: selectedDate });
+  const handleEndDateChange = (selectedDate: string) => {
+    updateField({ endDate: selectedDate });
   };
 
-  const handleStartTimeChange = (selectedTime: string) => {
-    // Validate if end time is before start time
-    if (endTime && selectedTime >= endTime) {
-      setTimeWarning("End time must be after start time.");
-    } else {
-      setTimeWarning(null);
-    }
+  useEffect(() => {
+    updateField({ endDate: startDate });
+  }, [startDate]);
 
+  const handleStartTimeChange = (selectedTime: string) => {
     updateField({ startTime: selectedTime });
   };
 
   const handleEndTimeChange = (selectedTime: string) => {
-    // Validate if end time is before start time
-    if (selectedTime <= startTime) {
-      setTimeWarning("End time must be after start time.");
+    updateField({ endTime: selectedTime });
+  };
+
+  useEffect(() => {
+    const currentDateTime = new Date();
+    const selectedStartDateTime = new Date(`${startDate}T${startTime}`);
+    const selectedEndDateTime = new Date(`${endDate}T${endTime}`);
+    console.log(startDate, startTime);
+
+    if (currentDateTime > selectedStartDateTime) {
+      setDateWarning("Event start date and time is in the past!");
+    } else {
+      setDateWarning(null);
+    }
+
+    if (selectedEndDateTime < selectedStartDateTime) {
+      setTimeWarning("Event must end after it starts!");
     } else {
       setTimeWarning(null);
```

Comment:
why do u pass in null here?

Reply:
so that there is no warning and if no warning then no flags raised

Reply:
put a TODO comment to make the null a variable with a more descriptive name and make it a jira ticket/find a better way to do this functionality without null.